### PR TITLE
install missing dependency imagemagick

### PR DESCRIPTION
**Pull request by hansvancalster**
pkgdown failed for #37. See https://github.com/inbo/protocolhelper/runs/2165908684?check_suite_focus=true

I am not familiar with github actions, but from the error I think this commit should solve it?

### CODE REVIEW

**Comment by hansvancalster** [comment]
Also, why does this error occur? The R `magick` package is listed in 'Suggests' not 'Imports'.

File: .github/workflows/pkgdown.yaml [hunk 1]
```diff
@@ -32,6 +32,7 @@ jobs:
 
       - name: Install dependencies
         run: |
+          brew imagemagick@6
           install.packages("remotes")
           remotes::install_deps(dependencies = TRUE)
```
Comment (ThierryO) [inline .github/workflows/pkgdown.yaml:37]: `remotes::install_deps(dependencies = TRUE)` will also install the `Suggests` packages.

**Comment by ThierryO** [comment]
This problem is solved in `checklist` workflows.

**Comment by hansvancalster** [comment]
> This problem is solved in `checklist` workflows.

Good to know. Is your suggestion I should use `checklist::check_package()`?

**Comment by ThierryO** [comment]
use `checklist::setup_package()`.

**Comment by hansvancalster** [comment]
So this PR is going to be a bit more elaborate than anticipated 😉 

Going down the rabbit hole of `checklist::check_package()`, which is really wonderful. However, some flagged issues ask me to change stuff which I don't want to do. For instance:

image

@ThierryO as you will agree, we should write `NEWS.md` with capital letters. Also changing stuff inside the `css` folder is I think something I should really not do. So, how should I deal with these issues?

**Comment by hansvancalster** [comment]
Also getting this note with `devtools::check()`:

image


which should also be ignored I guess? Via `checklist.yml` or is this something that the package should handle behind the scenes?

**Comment by ThierryO** [comment]
I'm writing a vignette on how to setup checklist with an existing package. Feedback is welcome: https://github.com/inbo/checklist/blob/78b0645dfe42de28ce2589a7b281b1600834ab0a/vignettes/introduction.Rmd#L19

**Comment by ThierryO** [comment]
Please run 

    remotes::install_github("inbo/checklist@documentation")
    checklist::setup_package()

That should fix the `.Rbuildignore` file and solve the `develtools::check()` errors.

A lot of the filename issues seem false positives. I'll look into that ASAP. (I'm having similar issues with inbo/INBOmd#69)

**Comment by hansvancalster** [comment]
Thanks! That fixed a lot of issues. The remaining once are:

````
####################################################################################################################

Checklist summary for the package located at:
C:/R/GitRepositories/protocolhelper

####################################################################################################################

2 new notes

--------------------------------------------------------------------------------------------------------------------
 checking CRAN incoming feasibility ... NOTE
Maintainer: 'Hans Van Calster <hans.vancalster@inbo.be>'

New submission

Version contains large components (0.1.9001) 
--------------------------------------------------------------------------------------------------------------------
 checking top-level files ... NOTE
Files 'README.md' or 'NEWS.md' cannot be checked without 'pandoc' being installed. 

####################################################################################################################

2 linters found.
`styler::style_file()` can fix some problems automatically. 

--------------------------------------------------------------------------------------------------------------------
1 times "functions should have cyclomatic complexity of less than 15, this has 21."
1 times "functions should have cyclomatic complexity of less than 15, this has 30."


####################################################################################################################


````

I can take care of the cyclomatic complexity. The note about 0.1.9001 however is problematic. Maybe I should change to 0.1.901 for development versions? R CMD CHECK will only complain if any of the components of version number is larger than 1234.

The note about pandoc is something you need to take care of in the checklist package?

**Comment by hansvancalster** [comment]
Running `checklist::check_package()` again gives a different result?

````
####################################################################################################################

Checklist summary for the package located at:
C:/R/GitRepositories/protocolhelper

####################################################################################################################

8 new warnings

--------------------------------------------------------------------------------------------------------------------
 Use `_` as separator in the basename of non-graphics files.
  File: `inst/css/fonts/FlandersArtSans-Bold.ttf` 
--------------------------------------------------------------------------------------------------------------------
 Use `_` as separator in the basename of non-graphics files.
  File: `inst/css/fonts/FlandersArtSans-Light.ttf` 
--------------------------------------------------------------------------------------------------------------------
 Use `_` as separator in the basename of non-graphics files.
  File: `inst/css/fonts/FlandersArtSans-Medium.ttf` 
--------------------------------------------------------------------------------------------------------------------
 Use `_` as separator in the basename of non-graphics files.
  File: `inst/css/fonts/FlandersArtSans-Regular.ttf` 
--------------------------------------------------------------------------------------------------------------------
 Use `_` as separator in the basename of non-graphics files.
  File: `inst/css/fonts/glyphicons-halflings-regular.ttf` 
--------------------------------------------------------------------------------------------------------------------
 Use `_` as separator in the basename of non-graphics files.
  File: `tests/testthat/test-add_subprotocols.R` 
--------------------------------------------------------------------------------------------------------------------
 Use `_` as separator in the basename of non-graphics files.
  File: `tests/testthat/test-insert_protocolsection.R` 
--------------------------------------------------------------------------------------------------------------------
 Use `_` as separator in the basename of non-graphics files.
  File: `tests/testthat/test-render_release.R` 

####################################################################################################################

2 new notes

--------------------------------------------------------------------------------------------------------------------
 checking CRAN incoming feasibility ... NOTE
Maintainer: 'Hans Van Calster <hans.vancalster@inbo.be>'

New submission

Version contains large components (0.1.9001) 
--------------------------------------------------------------------------------------------------------------------
 checking top-level files ... NOTE
Files 'README.md' or 'NEWS.md' cannot be checked without 'pandoc' being installed. 

####################################################################################################################

2 linters found.
`styler::style_file()` can fix some problems automatically. 

--------------------------------------------------------------------------------------------------------------------
1 times "functions should have cyclomatic complexity of less than 15, this has 21."
1 times "functions should have cyclomatic complexity of less than 15, this has 30."

####################################################################################################################

filename conventions: 11 errors
--------------------------------------------------------------------------------------------------------------------
 Folder names should only contain lower case letters, numbers and underscore.
They can start with a single dot.
Failing folder: `data-raw`
--------------------------------------------------------------------------------------------------------------------
 Basename must contain only lower case, numbers, `_` or `-`.
Fails: `.github/ISSUE_TEMPLATE.md`
--------------------------------------------------------------------------------------------------------------------
 Basename must contain only lower case, numbers, `_` or `-`.
Fails: `.github/SUPPORT.md`
--------------------------------------------------------------------------------------------------------------------
 Basename must contain only lower case, numbers, `_` or `-`.
Fails: `inst/css/fonts/FlandersArtSans-Bold.ttf`
--------------------------------------------------------------------------------------------------------------------
 Basename must contain only lower case, numbers, `_` or `-`.
Fails: `inst/css/fonts/FlandersArtSans-Light.ttf`
--------------------------------------------------------------------------------------------------------------------
 Basename must contain only lower case, numbers, `_` or `-`.
Fails: `inst/css/fonts/FlandersArtSans-Medium.ttf`
--------------------------------------------------------------------------------------------------------------------
 Basename must contain only lower case, numbers, `_` or `-`.
Fails: `inst/css/fonts/FlandersArtSans-Regular.ttf`
--------------------------------------------------------------------------------------------------------------------
 Basename must contain only lower case, numbers, `_` or `-`.
Fails: `inst/rmarkdown/templates/template_sfp_en/skeleton/NEWS.Rmd`
--------------------------------------------------------------------------------------------------------------------
 Basename must contain only lower case, numbers, `_` or `-`.
Fails: `inst/rmarkdown/templates/template_sfp_nl/skeleton/NEWS.Rmd`
--------------------------------------------------------------------------------------------------------------------
 Basename must contain only lower case, numbers, `_` or `-`.
Fails: `inst/rmarkdown/templates/template_spp_en/skeleton/NEWS.Rmd`
--------------------------------------------------------------------------------------------------------------------
 Basename must contain only lower case, numbers, `_` or `-`.
Fails: `inst/rmarkdown/templates/template_spp_nl/skeleton/NEWS.Rmd`


####################################################################################################################


````

**Comment by ThierryO** [comment]
I will not allow `.github/ISSUE_TEMPLATE.md`. Instead create the template as an md file in the `.github/ISSUE_TEMPLATE` folder. See https://docs.github.com/en/github/building-a-strong-community/about-issue-and-pull-request-templates

I'll allow the default filenames defined at https://docs.github.com/en/github/building-a-strong-community/creating-a-default-community-health-file#supported-file-types

**Comment by ThierryO** [comment]
Can you reminder me why you need `NEWS.Rmd` instead of `NEWS.md`

**Comment by hansvancalster** [comment]
> Can you reminder me why you need `NEWS.Rmd` instead of `NEWS.md`

See 60b29e5

**Comment by hansvancalster** [comment]
> I will not allow `.github/ISSUE_TEMPLATE.md`. Instead create the template as an md file in the `.github/ISSUE_TEMPLATE` folder. See https://docs.github.com/en/github/building-a-strong-community/about-issue-and-pull-request-templates
> 
> I'll allow the default filenames defined at https://docs.github.com/en/github/building-a-strong-community/creating-a-default-community-health-file#supported-file-types

See 731bf2c

**Comment by ThierryO** [comment]
@hansvancalster  The new check on filenames: inbo/checklist@e17c50afb. Can you test this?

**Comment by hansvancalster** [comment]
> @hansvancalster The new check on filenames: [inbo/checklist@e17c50a](https://github.com/inbo/checklist/commit/e17c50afb). Can you test this?

Works. The only flagged files are:

````
###################################################################################################################

3 new warnings

--------------------------------------------------------------------------------------------------------------------
 Use `_` as separator in the basename of non-graphics files.
  File: `tests/testthat/test-add_subprotocols.R` 
--------------------------------------------------------------------------------------------------------------------
 Use `_` as separator in the basename of non-graphics files.
  File: `tests/testthat/test-insert_protocolsection.R` 
--------------------------------------------------------------------------------------------------------------------
 Use `_` as separator in the basename of non-graphics files.
  File: `tests/testthat/test-render_release.R` 

````

**Comment by hansvancalster** [comment]
@ThierryO the issue with `Files 'README.md' or 'NEWS.md' cannot be checked without 'pandoc' being installed.` is a known Windows specific issue: https://github.com/r-lib/rcmdcheck/issues/109

The issue is still open. Is the suggestion to add these files to `.RBuildIgnore` a good one?

**Comment by hansvancalster** [comment]
>  Use `_` as separator in the basename of non-graphics files.
>   File: `tests/testthat/test-render_release.R` 

These files are created by `usethis::use_test()`, so the requirement to use `_` as separator in the basename of non-graphics files clashes with the filename created by this function.
I can rename, but compatibility of the `checklist` package with the `usethis` package is something to aim for. 
The preference for `_` probably comes from the preference to name R functions and objects following `snake_case` convention. However, for filenames and folder names, `-` and `_` separator would IMO and according to most best practices for file naming conventions both be OK (e.g. https://www.datanovia.com/en/blog/r-coding-style-best-practices/#file-naming-conventions).
A good example that might convince you is [how folder names are constructed in the protocols repo](https://github.com/inbo/protocols#folder-and-filename-syntaxis). This is an example where a mix of `_` and `-` makes sense, because `-` is used as a separator for `<shorttitle>` and `<protocolcode>` (e.g. sfp-401-en),  and `_` as a separator for the folder name constructed from both parts `<protocolcode>_<shorttitle>` (parse / regexp friendly).

**Comment by ThierryO** [comment]
I try to keep the number of exceptions as small as possible. Changing `test-` to `test_` is IMHO not a big deal for the user. Note that `usethis::use_vignette()` does not create a tidy `DESCRIPTION`. So `usethis` is fine to create a template. But the user still needs to  tweak a few things.

**Comment by hansvancalster** [comment]
> I try to keep the number of exceptions as small as possible. Changing `test-` to `test_` is IMHO not a big deal for the user. Note that `usethis::use_vignette()` does not create a tidy `DESCRIPTION`. So `usethis` is fine to create a template. But the user still needs to tweak a few things.

OK, I've renamed the tests. I can live with this for package checks, but I would definitely be in favour of relaxing this requirement for people who want to use `checklist::check_filename()` on repo's that are not package repo's. That would mean broadening the scope of the function.  I can image that you already thought about something similar to `check_package()` but for non-package repo's `check_repo()`.

**Comment by hansvancalster** [comment]
@ThierryO Can you take a look at the failing checks?
When I run checklist::check_package() locally, everything is fine, but here I see again "Configuration failed to find the Magick++ library. ". 

The checks on Ubuntu fail because the package can't be installed. Can you test this on your system?

**Comment by hansvancalster** [comment]
CI `check_package` throws a warning "no visible global function definition for 'shell'", which is a function used in the `protocolhelper::execshell` function. 
On a Windows machine, this warning does not occur because the note in the help `?base::shell()` mentions: 'this function exists only on Windows: on all other platform system uses a shell.'

So despite the fact that this is acknowledged in the function:

```
execshell <- function(commandstring, intern = FALSE, ...) {
  if (.Platform$OS.type == "windows") {
    res <- shell(commandstring, intern = TRUE, ...)
  } else {
    res <- system(commandstring, intern = TRUE, ...)
  }
  if (!intern) {
    if (length(res) > 0) cat(res, sep = "\n") else return(invisible())
    } else return(res)
}
```

The `lintr` check is not smart enough to detect that `base::shell()` will not be used on non-Windows OS. 
I don't see a solution to solve this. Any ideas? @ThierryO  @florisvdh

**Comment by ThierryO** [comment]
Add `# nolint` at the end of the line using `shell`. Then lintr will ignore the line. Use this only in case you can't fix it.

I'm wondering why you're passing git commands to the shell. `git2r` has R functions for all those commands.

**Comment by hansvancalster** [comment]
> Add `# nolint` at the end of the line using `shell`. Then lintr will ignore the line. Use this only in case you can't fix it.
>

Great! That will fix it.
 
> I'm wondering why you're passing git commands to the shell. `git2r` has R functions for all those commands.

We are using `git show` among others via the shell. Neither `git2r` nor `gert` has an R equivalent for that command.

**Pull request merged**